Core/NoOpRunner.Core/GameFrameCycleFacade.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+using NoOpRunner.Core.Decorators;
+using NoOpRunner.Core.Enums;
+using NoOpRunner.Core.Interfaces;
+using NoOpRunner.Core.Rendering;
+
+namespace NoOpRunner.Core
+{
+    public class GameFrameCycleFacade
+    {
+        private IVisualElement Player { get; set; }
+        
+        private IList<PowerUps> DisplayingPlayerOnePowerUps { get; set; }
+
+        public GameFrameCycleFacade()
+        {
+            DisplayingPlayerOnePowerUps = new List<PowerUps>();
+        }
+
+        public async Task HostGameCycle(NoOpRunner game, Canvas playerCanvas, Canvas powerUpsCanvas,
+            Canvas platformsCanvas)
+        {
+
+            game.Player.OnLoopFired((WindowPixel[,]) game.PlatformsContainer.GetShapes().Clone());
+
+            BaseCycle(game, playerCanvas, platformsCanvas, powerUpsCanvas);
+
+            await game.UpdateClientsGame();
+        }
+
+        public void ClientGameCycle(NoOpRunner game, Canvas playerCanvas, Canvas platformsCanvas, Canvas powerUpsCanvas)
+        {
+            //More incoming
+            BaseCycle(game, playerCanvas, platformsCanvas, powerUpsCanvas);
+        }
+
+        private void BaseCycle(NoOpRunner game, Canvas playerCanvas, Canvas platformsCanvas, Canvas powerUpsCanvas)
+        {
+            if (Player == null)
+            {
+                Player = game.Player;
+            }
+            
+            game.Player.LoopPowerUps();
```

Comment:
If you assign Player variable then just use that instead of game.Player

Reply:
That's old version.

Reply:
And why do we need two versions

Reply:
no i fixed this yesterday, idk why it show you this